src/Event/ReturnEvent.php

```diff
@@ -25,13 +27,31 @@ class ReturnEvent extends Event
     private $return;
 
     /**
+     * The request sent to the client
+     *
+     * @var RequestInterface
+     */
+    private $request;
+
+    /**
+     * The response from the client
+     *
+     * @var ResponseInterface
+     */
+    private $response;
+
+    /**
      * Constructor
      *
      * @param mixed $return
+     * @param RequestInterface|null $request
+     * @param ResponseInterface|null $response
      */
-    public function __construct($return)
+    public function __construct($return, RequestInterface $request = null, ResponseInterface $response = null)
```

Comment:
I wanted to maintain BC but also I wasn't 100% positive that the response would not be null in some cases, such as async requests.